### PR TITLE
fix: 2fa authentication error handling

### PR DESCRIPTION
**Pull request by thiessenp-cds**
# Summary | Résumé

Adds more error handling to the Verify and Re-Verify (send a new code) screens.

Test by signing a user in entering a code. Also try the "send a new code".

What's left:
Add more errors to ReVerify one the end point `/api/auth/2fa/request-new-verification-code` is more complete

### CODE REVIEW

**Comment by timarney** [comment]
A few things for a follow-up PR

We should clear the previous error state when coming back from re-send.

Screenshot 2023-06-08 at 9 31 09 AM

Also thinking we should also clear the toast etc... when pressing submit for a valid code.

File: components/auth/ReVerify.tsx [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+          "X-CSRF-Token": token,
+        },
+        data: new URLSearchParams({
+          email: username.current,
+          authenticationFlowToken: authenticationFlowToken.current,
+        }),
+        timeout: process.env.NODE_ENV === "production" ? 60000 : 0,
+      });
+
+      // TODO: update to match any newe rrors API sends the future
```
Comment (timarney) [inline components/auth/ReVerify.tsx:52]:
```suggestion
      // TODO: update to match any new errors API sends the future
```

File: components/auth/Verify.tsx [hunk 2]
```diff
@@ -137,12 +170,11 @@ export const Verify = ({ username, authenticationFlowToken }: VerifyProps): Reac
                 {t("verify.confirmButton")}
               </Button>
               <div className="flex mt-12">
+                {/* Note: this is a special button that looks like a link with specific styling. Otherwise the Button component should be used */}
                 <button
```
Comment (timarney) [inline components/auth/Verify.tsx:174]:
```suggestion
                <Button
                  theme="link"
                  onClick={() => setIsReverify(true)}
                >
                  {t("verify.resendConfirmationCodeButton")}
                </Button>{" "}
```

File: components/auth/Verify.tsx [hunk 3]
```diff
@@ -137,12 +170,11 @@ export const Verify = ({ username, authenticationFlowToken }: VerifyProps): Reac
                 {t("verify.confirmButton")}
               </Button>
               <div className="flex mt-12">
+                {/* Note: this is a special button that looks like a link with specific styling. Otherwise the Button component should be used */}
```
[the remark this answers is not in the document]
Comment (thiessenp-cds) [reply components/auth/Verify.tsx:173]: The link hover and removing an underline what bugged me about style if I remember right. But close enough to a link, will do

[the remark this answers is not in the document]
Comment (timarney) [reply components/auth/Verify.tsx]: > The link hover and removing an underline what bugged me about style 

In which case we should look to update across the board -- in a fresh issue :)

**Review by timarney — APPROVED** [review]

**Pull request merged**